libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/token/jwts/JwtsSigningKeyResolver.java

```diff
@@ -45,6 +45,9 @@ public class JwtsSigningKeyResolver implements SigningKeyResolver {
     private static final ObjectMapper JSON_MAPPER = initJsonMapper();
     private final SSLContext sslContext;
     private final String jwksUri;
+    private final boolean skipConfig;
+    private static long lastZtsJwkFetchTime;
+    private static long millisBetweenZtsCalls;
```

Comment:
what is the default value for this setting? this change would require every user of the library to set the value otherwise with unknown key ids the library will be making a call to the server every time. we can probably start with a conserve once a day value.

Reply:
The default value is set to 30 minutes. This setting can be modified in the property file.

https://github.com/AthenZ/athenz/blob/e1e5e5ff152422b1a1b8bcd2249f23c2761ddccf/clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java#L182

In order to avoid increasing the server load by sending requests to ZTS every time an unknown key ID is specified, we have set it to make a request every 30 minutes.

Reply:
you're incorrectly assuming that it's the only place that the library is used. it's not a server library. it's a common client shared library so there are other applications that are using that functionality.

Reply:
I apologize for the mistake. There was an error in the comment. It's not a property file, but a system property.
When need to modify the settings while using the client library, can make use of system properties.

Reply:
Doesn't matter if it can be setup using a system property. We cannot make changes in shared client libraries where we change its current behavior and technically could result in DOS attacks. You must set the default value for the millisBetweenZtsCalls field to one day and then you can change that value where it's needed (e.g. with zpu)